x-pack/legacy/plugins/ml/public/components/navigation_menu/navigation_menu.tsx

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Fragment, SFC, useState } from 'react';
+import { EuiFlexGroup, EuiFlexItem, EuiTabs, EuiTab } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import chrome from 'ui/chrome';
+import { TopNav } from './top_nav';
+
+interface Tab {
+  id: string;
+  name: any;
+  disabled: boolean;
+}
+
+interface Props {
+  dateFormat: string;
+  disableLinks: boolean;
+  showTabs: boolean;
+  tabId: string;
+  timeHistory: any;
+  timefilter: any;
+}
+
+function moveToSelectedTab(selectedTabId: string) {
+  window.location.href = `${chrome.getBasePath()}/app/ml#/${selectedTabId}`;
+}
+
+function getTabs(disableLinks: boolean): Tab[] {
+  return [
+    {
+      id: 'jobs',
+      name: i18n.translate('xpack.ml.navMenu.jobManagementTabLinkText', {
+        defaultMessage: 'Job Management',
+      }),
+      disabled: disableLinks,
+    },
+    {
+      id: 'explorer',
+      name: i18n.translate('xpack.ml.navMenu.anomalyExplorerTabLinkText', {
+        defaultMessage: 'Anomaly Explorer',
+      }),
+      disabled: disableLinks,
+    },
+    {
+      id: 'timeseriesexplorer',
+      name: i18n.translate('xpack.ml.navMenu.singleMetricViewerTabLinkText', {
+        defaultMessage: 'Single Metric Viewer',
+      }),
+      disabled: disableLinks,
+    },
+    {
+      id: 'data_frames',
+      name: i18n.translate('xpack.ml.navMenu.dataFrameTabLinkText', {
+        defaultMessage: 'Data Frames',
+      }),
+      disabled: false,
+    },
+    {
+      id: 'datavisualizer',
+      name: i18n.translate('xpack.ml.navMenu.dataVisualizerTabLinkText', {
+        defaultMessage: 'Data Visualizer',
+      }),
+      disabled: false,
+    },
+    {
+      id: 'settings',
+      name: i18n.translate('xpack.ml.navMenu.settingsTabLinkText', {
+        defaultMessage: 'Settings',
+      }),
+      disabled: disableLinks,
+    },
+  ];
+}
+
+export const NavigationMenu: SFC<Props> = ({
+  dateFormat,
+  disableLinks,
+  showTabs,
+  tabId,
+  timeHistory,
+  timefilter,
+}) => {
+  const [tabs] = useState(getTabs(disableLinks));
```

Comment:
Since there's no `setTabs`, IMHO we don't require `useState()` for this and you could just do `const tabs = getTabs(disableLinks)`. This version using `useState` would actually never pick up if the prop `disableLinks` changed on a re-render, it would always pick the value on first render as the default.

Reply:
Not relevant anymore since tabs have been moved to own component 999248fd7c9196611a3c41fb091ed4d4cc8f8f23